src/librustc_typeck/check/regionck.rs

```diff
@@ -899,10 +896,6 @@ impl<'a, 'tcx> RegionCtxt<'a, 'tcx> {
             }
 
             cmt = self.with_mc(|mc| mc.cat_expr_adjusted(expr, cmt, &adjustment))?;
-
-            if let Categorization::Deref(_, mc::BorrowedPtr(_, r_ptr)) = cmt.cat {
-                self.mk_subregion_due_to_dereference(expr.span, expr_region, r_ptr);
```

Comment:
do you know what this was here for? I just want a note in this PR about why we don't need it, no need to add comments to the soruce code or anything.

Reply:
(is this a case where mir-borrowck (and/or nll/type-check) is imposing the same constraint and so we do not need to impose it here?)

Reply:
That's the case for most of the constraints in this file. This check specifically is to ensure, in the language of NLL, that `'a` is live at the point marked:

```rust
let x: &'a (i32, i32) = ...;
let _ = x.0; //here
```

This is redundant because:
* There's no longer any check that makes use of ReScopes (except for the leak check, but that's hopefully going away soon).
* NLL does is own region liveness inference.
* I think that we're already adding this relation because all of the regions in the unadjusted expression are already marked live.